### PR TITLE
Code Refactor

### PR DESCRIPTION
**Pull request by imanghafoori1**
Hi, I am a rust learner and I stumbled upon your video on youtube and reached your repo on github.
While I was reading and tampering with your code to learn, I just made it a little bit DRY er.
I think it would be nice for other people like me, since it makes it easier to read and learn.

Thank you about your tutorials.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -238,11 +230,29 @@ fn create_tile_scroll(sprite: Sprite, x: f32, jump: f32) -> Entity {
...
+    let total_dist = width * (NUMBER_OF_TILES as f32);
+    (0..NUMBER_OF_TILES)
         .into_iter()
-        .map(|i| create_tile_scroll(sprite.clone(), (i as f32) * (width as f32), total_dist))
+        .map(|i| create_tile_scroll(sprite.clone(), (i as f32) * width, total_dist))
         .collect()
 }
+
+impl GameState {
+    fn handle_loosing(&mut self, ctx: &mut Context, state: PlayState) {
```
Comment (AndrewJakubowicz) [inline src/main.rs]: Is this a tiny typo? `handle_losing`?

Also this PR is amazing.

**Comment by imanghafoori1** [comment]
although it is not perfect yet, but at least, a bit better than before.
and a sweet experiment for me.

**Comment by imanghafoori1** [comment]
I am planning to massage this code base a lot more.
I branch off from here, so the rest of the changes will be included in the next PR

**Comment by AndrewJakubowicz** [comment]
This is awesome!

**Pull request merged**